schema/shared/src/main/scala/zio/blocks/schema/binding/RegisterPool.scala

```diff
@@ -1,22 +1,29 @@
 package zio.blocks.schema.binding
 
 final class RegisterPool private () {
-  private[this] var registers: Array[Registers] = Array.fill[Registers](8)(Registers(RegisterOffset.Zero))
-  private[this] var used: Int                   = 0
+  private[this] var used: Int = 0
+  private[this] var weakRef: java.lang.ref.WeakReference[Array[Registers]] =
+    new java.lang.ref.WeakReference(Array(allocate()))
```

Comment:
Is the overhead of `WeakReference` really worth the memory saving?

If object registers are 'zeroed out', we can collect the things they reference, without collecting the registers themselves, which might be a nice midway point.